parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -123,10 +123,19 @@ public class ParquetFileReader implements Closeable {
 
   public static String PARQUET_READ_PARALLELISM = "parquet.metadata.read.parallelism";
 
+  private ParquetMetricsCallback metricsCallback;
+
   private final ParquetMetadataConverter converter;
 
   private final CRC32 crc;
 
+  /**
+   * set a callback to send back metrics info
+   */
+  public void initMetrics(ParquetMetricsCallback callback) {
```

Comment:
Why not set the `callback` in `ParquetReadOptions`?

Reply:
That's a good idea, but it may require changes to some of the public constructors which are in use by others. For instance Spark has this https://github.com/apache/spark/blob/15e99cf676d9de02c54ca5ebe9a2bc6a3ce014e5/sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/SpecificParquetRecordReaderBase.java#L104
In this case, the constructor is deprecated and it looks like Spark could benefit from a constructor that takes a ParquetReadOptions as a parameter. I can introduce a new constructor but wasn't sure that we would want to do so. Having an additional API to set the metrics callback seemed like the simpler option.
WDYT?

Reply:
@ConeyLiu Gentle ping, do you have any comment?

Reply:
Sorry for missing this. https://github.com/apache/parquet-mr/blob/afd39dde8fd762bf696fea3dab16d45eae1093c3/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java#L669
The `open` method passing a `footer` is deprecated. I think we should support it with `ParquetReadOptions`. Then Spark could benefit from this as well. 

I am OK with the current changes to unblock this PR.